### PR TITLE
✨ Idle timer: close study after 30'

### PR DESCRIPTION
**Pull request by odeimaiz**
## What do these changes do?

This feature is only enabled for the ```s4llite``` product

In order to free resources, when a user has a study open and there is no activity in the osparc tab (```mousemove```, ```mousedown``` and ```keydown```), the frontend will close the study.

After 15', a warning message with a countdown will pop up saying: ```Are you still there? If not, the project will be closed in XX:XX```. When the countdown goes down to 0, the study will get closed.

Demo with timeout set to 20" and warning to 10":
IdleTimeout

## Related issue/s

Idle timeout: ITISFoundation/osparc-issues#765

### CODE REVIEW

File: services/static-webserver/client/source/class/osparc/utils/Utils.js [hunk 1]
```diff
@@ -304,6 +304,14 @@ qx.Class.define("osparc.utils.Utils", {
       return osparc.utils.Utils.formatDate(value) + " " + osparc.utils.Utils.formatTime(value);
     },
 
+    formatSeconds: function(seconds) {
+      const min = Math.floor(seconds / 60);
+      const sec = seconds - min * 60;
```
Comment (pcrespov) [inline services/static-webserver/client/source/class/osparc/utils/Utils.js:309]:
```javascript
const minutes = Math.floor(seconds / 60);
const remainingSeconds = seconds % 60;
```

**Review by pcrespov — APPROVED** [review]

**Review by GitHK — APPROVED** [review]
👍 very nice

**Review by elisabettai — APPROVED** [review]
Very very nice!

Suggestion: someone might understand that there's an absolute max time they're allowed to use the project, so I would clarify that it's only if idle.

The message can become something like  "The project will be closed in ... due to inactivity. If you're still using it you can ignore this message."

**Comment by odeimaiz** [comment]
> The message can become something like "The project will be closed in ... due to inactivity. If you're still using it you can ignore this message."

If you are using it, you won't get the message.
- The timer is started when the inactivity starts
- 15' later you will get the message
- 30' after the inactivity timer started and if you are still inactive, the study gets closed

**Comment by elisabettai** [comment]
> > The message can become something like "The project will be closed in ... due to inactivity. If you're still using it you can ignore this message."
> 
> If you are using it, you won't get the message.
> 
>     * The timer is started when the inactivity starts
> 
>     * 15' later you will get the message
> 
>     * 30' after the inactivity timer started and if you are still inactive, the study gets closed

That's clear. 

If the user comes back after 20' from the start of the inactivity, I suppose he will see the message.
At that point he might resume working on the project again and the timer is reset, I suppose. In this case I was thinking it might be worth notifyin him that he can resume working without being worried that the project will be closed after 10'.

**Review by sanderegg — APPROVED** [review]

**Pull request merged**